src/na/device/Generator.cpp

```diff
@@ -0,0 +1,410 @@
+       << operation.fidelity() << "}));";
+  }
+  for (const auto& operation : device.shuttling_units()) {
+    os << "\\\n"
+          "  var.emplace_back(std::make_unique<MQT_NA_QDMI_Operation_impl_d>("
+          "MQT_NA_QDMI_Operation_impl_d{\""
+       << operation.name() << " (Load)\", OperationType::SHUTTLING_LOAD, "
+       << operation.num_parameters() << ", 0, "
+       << static_cast<double>(operation.load_duration()) * timeUnit << ", "
+       << operation.load_fidelity() << "}));";
+    os << "\\\n"
+          "  var.emplace_back(std::make_unique<MQT_NA_QDMI_Operation_impl_d>("
+          "MQT_NA_QDMI_Operation_impl_d{\""
+       << operation.name() << " (Move)\", OperationType::SHUTTLING_MOVE, "
+       << operation.num_parameters() << ", 0, 0, 0}));";
+    os << "\\\n"
+          "  var.emplace_back(std::make_unique<MQT_NA_QDMI_Operation_impl_d>("
+          "MQT_NA_QDMI_Operation_impl_d{\""
+       << operation.name() << " (Store)\", OperationType::SHUTTLING_STORE, "
+       << operation.num_parameters() << ", 0, "
+       << static_cast<double>(operation.store_duration()) * timeUnit << ", "
+       << operation.store_fidelity() << "}));";
+  }
+  os << "\n";
+}
+
+/**
+ * @brief Writes the decoherence times from the Protobuf message.
+ * @param device The Protobuf message containing the device configuration.
+ * @param timeUnit The time unit to use for the decoherence times.
+ * @param os The output stream to write the sites to.
+ */
+auto writeDecoherenceTimes(const Device& device, const double timeUnit,
+                           std::ostream& os) -> void {
+  os << "#define INITIALIZE_T1(var) var = "
+     << static_cast<double>(device.decoherence_times().t1()) * timeUnit
+     << ";\n";
+  os << "#define INITIALIZE_T2(var) var = "
+     << static_cast<double>(device.decoherence_times().t2()) * timeUnit
+     << ";\n";
+}
+} // namespace
+
+auto writeJSONSchema(std::ostream& os) -> void {
+  // Create a default device configuration
+  Device device;
+
+  // Fill each repeated field with an empty message
+  populateRepeatedFields(&device);
+
+  // Set print options
+  google::protobuf::util::JsonPrintOptions options;
+  options.always_print_fields_with_no_presence = true;
+
+  // Convert to JSON
+  std::string json;
+  const auto status =
+      google::protobuf::util::MessageToJsonString(device, &json, options);
+  if (!status.ok()) {
+    std::stringstream ss;
+    ss << "Failed to convert Protobuf message to JSON: " << status.ToString();
+    throw std::runtime_error(ss.str());
+  }
+
+  // Write to output stream
+  os << json;
+}
+
+auto writeJSONSchema(const std::string& path) -> void {
+  // Write to file
+  std::ofstream ofs(path);
+  if (ofs.is_open()) {
+    writeJSONSchema(ofs);
+    ofs.close();
+    SPDLOG_INFO("JSON template written to {}", path);
+  } else {
+    std::stringstream ss;
+    ss << "Failed to open file for writing: " << path;
+    throw std::runtime_error(ss.str());
+  }
```

Comment:
Truly minor and definitely a stylistic choice, but I would prefer the pattern
```c++
  std::ofstream ofs(path);
  if (!ofs.is_good()) {
    ... error ...
  }
  ... proceed with file ...
```
similar to how it is used in the read function down below.